### PR TITLE
fix: send proper status to server in cm360

### PR DESCRIPTION
**Pull request by aashishmalik**
## What are the changes introduced in this PR?

 send proper status to server in cm360

## What is the related Linear task?

Resolves INT-1509

## Please explain the objectives of your changes below

Put down any required details on the broader aspect of your changes. If there are any dependent changes, **mandatorily** mention them here

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### Any technical or performance related pointers to consider with the change?

N/A

@coderabbitai review

<hr>

### Developer checklist

- [ ] My code follows the style guidelines of this project

- [ ] **No breaking changes are being introduced.**

- [ ] All related docs linked with the PR?

- [ ] All changes manually tested?

- [ ] Any documentation changes needed with this change?

- [ ] Is the PR limited to 10 file changes?

- [ ] Is the PR limited to one linear task?

- [ ] Are relevant unit and component test-cases added?

### Reviewer checklist

- [ ] Is the type of change in the PR title appropriate as per the changes?

- [ ] Verified that there are no credentials or confidential data exposed with the changes.

### CODE REVIEW

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3127/test-report.html

File: test/integrations/destinations/campaign_manager/dataDelivery/oauth.ts [hunk 1]
```diff
@@ -326,14 +326,14 @@ export const v1oauthScenarios: ProxyV1TestData[] = [
     },
     output: {
       response: {
-        status: 500,
+        status: 401,
```
Comment (koladilip) [inline test/integrations/destinations/campaign_manager/dataDelivery/oauth.ts:329]: will this be retried or aborted by rudder server and is that expected

File: test/integrations/destinations/campaign_manager/dataDelivery/oauth.ts [hunk 2]
```diff
@@ -389,14 +389,14 @@ export const v1oauthScenarios: ProxyV1TestData[] = [
     },
     output: {
       response: {
-        status: 500,
+        status: 403,
```
Comment (koladilip) [inline test/integrations/destinations/campaign_manager/dataDelivery/oauth.ts:392]: will this be retried or aborted by rudder server and is that expected

**Review by koladilip — APPROVED** [review]

File: test/integrations/destinations/campaign_manager/dataDelivery/oauth.ts (hunk 1 above)
Comment (aashishmalik) [reply test/integrations/destinations/campaign_manager/dataDelivery/oauth.ts:329]: authcategory is `REFRESH_TOKEN` , so retired

File: test/integrations/destinations/campaign_manager/dataDelivery/oauth.ts (hunk 2 above)
Comment (aashishmalik) [reply test/integrations/destinations/campaign_manager/dataDelivery/oauth.ts:392]: here auth category is `incative`..so these won't be retried

File: test/integrations/destinations/campaign_manager/dataDelivery/oauth.ts (hunk 1 above)
Comment (koladilip) [reply test/integrations/destinations/campaign_manager/dataDelivery/oauth.ts:329]: so 4xx status will be handled properly right in rudder server

Comment (utsabc) [reply test/integrations/destinations/campaign_manager/dataDelivery/oauth.ts:329]: @aashishmalik did you test this behaviour with server. Lets test this scenario buy running transformer and server and running debug-mode on transformer and changing the response in proxy

Comment (utsabc) [reply test/integrations/destinations/campaign_manager/dataDelivery/oauth.ts:329]: For refresh ideally we should send 500

Comment (aashishmalik) [reply test/integrations/destinations/campaign_manager/dataDelivery/oauth.ts:329]: updated in linear, server did refresh call

**Pull request merged**